### PR TITLE
Use named routes instead url_for

### PR DESCRIPTION
**Pull request by seuros**
From the docs, it seem we using named routes instead of url_for

### CODE REVIEW

File: railties/lib/rails/generators/test_unit/controller/templates/functional_test.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ class <%= class_name %>ControllerTest < ActionDispatch::IntegrationTest
 <% else -%>
 <% actions.each do |action| -%>
   test "should get <%= action %>" do
-    get url_for(action: :<%= action %>)
+    get <%= action %>_<%= plural_table_name %>_path
```
Comment (seuros) [inline railties/lib/rails/generators/test_unit/controller/templates/functional_test.rb]: this will fail for :show and :delete and :index.

**Comment by gsamokovarov** [comment]
Hey, since we seemed to raced on this one, here is the diff that I had. Hopefully it can help you flesh out some of those failures:

https://github.com/rails/rails/compare/master...gsamokovarov:controller-default-integration-test?expand=1

**Comment by eileencodes** [comment]
Hey @seuros I reverted the other commit because I didn't want to leave the build broken. There's one last failure (it looks to be order dependent because I could only get it to fail after about 5 runs of that file).  Here's a link to the failure. https://travis-ci.org/rails/rails/jobs/96655073#L3295 I'm not sure what's up and why this test is failing.

**Comment by eileencodes** [comment]
Since I reverted the other one, when you figure out the new failures can you update the commit message to say what you're changing and that it's because ActionController::TestCase is deprecated in favor of ActionDispatch::IntegrationTest?

**Comment by eileencodes** [comment]
Closed in https://github.com/rails/rails/commit/98a039dcd987fd3a8c2c0cbd527c71ac3ffaaf06 and https://github.com/rails/rails/commit/becf1bcb1d6e72e782995507ebc89d75db357184. I made some updates to the code, rewrote the commit message and figured out why that one test was failing. I saved pushed the first one as your commit from here. Thanks for working on this!

**Pull request closed without merging**